drgn_tools/dm.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+"""
+Helpers for device mapper devices.
+"""
+import argparse
+from typing import Iterable
+from typing import Tuple
+
+from drgn import Object
+from drgn import Program
+from drgn.helpers.linux.list import list_for_each_entry
+from drgn.helpers.linux.rbtree import rbtree_inorder_for_each_entry
+
+from drgn_tools.corelens import CorelensModule
+from drgn_tools.module import ensure_debuginfo
+from drgn_tools.table import print_table
+from drgn_tools.util import BitNumberFlags
+from drgn_tools.util import kernel_version
+from drgn_tools.util import variable_exists
+
+
+def for_each_dm_hash(prog: Program) -> Iterable[Tuple[Object, str]]:
+    # bucket size is 64
+    for i in range(64):
+        head = prog["_name_buckets"][i]
+        for hc in list_for_each_entry(
+            "struct hash_cell", head.address_of_(), "name_list"
+        ):
+            yield hc.md, hc.name.string_().decode()
```

Comment:
The declaration of `_name_buckets` is an explicitly-sized array:
```c
static struct list_head _name_buckets[NUM_BUCKETS];
```

So drgn already knows the size of the array and you can just iterate over it. No need to hardcode the `NUM_BUCKETS` value.

```suggestion
    for head in prog["_name_buckets"]:
        for hc in list_for_each_entry(
            "struct hash_cell", head.address_of_(), "name_list"
        ):
            yield hc.md, hc.name.string_().decode()
```